docs/data/date-pickers/fields/ControlledSelectedSectionsSingleInputRangeField.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+import Button from '@mui/material/Button';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import {
+  MuiDateSectionName,
+  FieldSelectedSections,
+  FieldRef,
+  RangeFieldSection,
+  RangePosition,
+} from '@mui/x-date-pickers-pro';
+import { SingleInputDateRangeField } from '@mui/x-date-pickers-pro/SingleInputDateRangeField';
+
+export default function ControlledSelectedSectionsSingleInputRangeField() {
+  const [selectedSections, setSelectedSections] =
+    React.useState<FieldSelectedSections>(null);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const fieldRef = React.useRef<FieldRef<RangeFieldSection>>(null);
+
+  const setSelectedDateSectionName = (
+    selectedDateSectionName: MuiDateSectionName,
+    position: RangePosition,
+  ) => {
+    if (!fieldRef.current) {
+      return;
+    }
+
+    inputRef.current?.focus();
+    const sections = fieldRef.current.getSections().map((el) => el.dateSectionName);
+    setSelectedSections(
+      position === 'start'
+        ? sections.indexOf(selectedDateSectionName)
+        : sections.lastIndexOf(selectedDateSectionName),
+    );
+  };
+
+  const renderDateHeader = (position: RangePosition) => (
+    <Stack spacing={2} alignItems="center">
+      <Typography textTransform="capitalize">{position}</Typography>
+      <Stack direction="row" spacing={1}>
+        {(['month', 'day', 'year'] as const).map((sectionName) => (
+          <Button
+            size="small"
+            variant="outlined"
+            onClick={() => setSelectedDateSectionName(sectionName, position)}
+          >
+            {sectionName}
+          </Button>
+        ))}
+      </Stack>
+    </Stack>
+  );
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <Stack spacing={2}>
```

Comment:
Could we make sure that we are not introducing new mobile layout regressions, especially when Olivier has made such a nice effort fixing mostly all of them..? 🙈 🤔 
![Screenshot 2023-02-24 at 10 23 49](https://user-images.githubusercontent.com/4941090/221129319-de2c30cf-712d-4175-9d04-77895eaba4c1.png)

Reply:
Fixed :+1:

Reply:
Multi input variant is also begging for this change. 😱